rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/validation/CachingValidator.java

```diff
@@ -128,12 +134,12 @@ protected boolean validateCachingBehavior()
           Ints.checkedCast(readSize),
           fileLength,
           fileLastModified,
-          VALIDATOR_CLUSTER_TYPE);
+          VALIDATOR_CLUSTER_TYPE).isStatus();
       if (!dataRead) {
         return false;
       }
 
-      locations = bookKeeper.getCacheStatus(request);
+      locations = bookKeeper.getCacheStatus(request).getBlocks();
```

Comment:
why do we need to check this again? I am missing something?

Reply:
readData is called before it.